x/supply/query/querier.go

```diff
@@ -0,0 +1,68 @@
+package query
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	vestingexported "github.com/cosmos/cosmos-sdk/x/auth/vesting/exported"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/supply"
+	"github.com/ovrclk/akash/sdkutil"
+	"github.com/ovrclk/akash/x/supply/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/exported"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// NewQuerier creates and returns a new supply querier instance
+func NewQuerier(cdc *codec.Codec, accKeeper types.AccountKeeper, supKeeper types.SupplyKeeper) sdk.Querier {
+	return func(ctx sdk.Context, path []string, req abci.RequestQuery) (res []byte, err error) {
+		switch path[0] {
+		case circulatingPath:
+			return queryCirculatingSupply(ctx, cdc, accKeeper, supKeeper)
+		default:
+			return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unknown query for endpoint: %s", path[0])
+		}
+	}
+}
+
+func queryCirculatingSupply(ctx sdk.Context, cdc *codec.Codec, accKeeper types.AccountKeeper,
+	supKeeper types.SupplyKeeper) (res []byte, err error) {
+	var supplyData Supply
+	var totalOriginal, totalVesting sdk.Coins
+
+	initialSupply := sdk.NewCoins(sdk.NewCoin("uakt", sdk.NewInt(100000000000000)))
```

Comment:
May be we should read this from supplyKeeper itself. Can we have something like, `getSupply` at `height:1`

Reply:
Agree, we shouldn't hard-code this.

But I think that we shouldn't be concerned about initial supply - iterating through accounts should give us the info we need.

I do think that we'll have to look into the specific account types rather than using just using `exported.Account`, though.